AuthSamples/SwiftSample/ViewController.swift

```diff
@@ -342,12 +342,25 @@ final class ViewController: UIViewController, UITextFieldDelegate {
     providerListLabel.text = providers?.joined(separator: ", ")
     if let photoURL = user?.photoURL {
       lastPhotoURL = photoURL
-      DispatchQueue.global(priority: DispatchQueue.GlobalQueuePriority.background).async {
-        if let imageData = try? Data(contentsOf: photoURL) {
-          let image = UIImage(data: imageData)
-          DispatchQueue.main.async {
-            if self.lastPhotoURL == photoURL {
-              self.profileImage.image = image
+      if #available(iOS 8.0, *) {
+        DispatchQueue.global(qos: .background).async {
```

Comment:
Instead of wrapping the whole dispatch statement, you can just wrap the declaration of the queue if that's the only deprecated API.

```swift
let queue: DispatchQueue
if #available(iOS 8.0, *) {
  queue = DispatchQueue.global(qos: .background)
} else {
  queue = DispatchQueue.global(priority: DispatchQueue.GlobalQueuePriority.background)
}

queue.async { /* ... */ }
```